src/auth/detectProtocolLaunch.ts

```diff
@@ -0,0 +1,180 @@
+import { checkWindowAPI, BLOCKSTACK_HANDLER } from '../utils'
+import { Logger } from '../logger'
+
+/**
+ * Detects if the native auth-browser is installed and is successfully 
+ * launched via a custom protocol URI. 
+ * @param {String} authRequest
+ * The encoded authRequest to be used as a query param in the custom URI. 
+ * @param {String} successCallback
+ * The callback that is invoked when the protocol handler was detected. 
+ * @param {String} failCallback
+ * The callback that is invoked when the protocol handler was not detected. 
+ * @return {void}
+ */
+export function detectProtocolLaunch(
```

Comment:
Note to reviewers: this `detectProtocolLaunch` function had no code/logic changes. It was simply moved to its own file. Due to its monolithic nature, and it being all DOM code, which doesn't make sense in context of the other functions in the file it was previously in.